### PR TITLE
Add unit tests

### LINKED ISSUE #26: Write Unit Tests
#### **Description:**
write comprehensive unit tests for the existing functionality of the LibraryMan project. Unit testing is crucial to ensure that the individual components are functioning correctly and to catch potential bugs early in the development process.

#### **Acceptance Criteria:**

1. Each function should have a minimum of 80% code coverage.
2. Tests should cover edge cases and possible failure scenarios.
3. All unit tests must be executable through the project's CI pipeline. (Optional)
4. Documentation for how to run the tests locally should be updated if necessary.

#### **Note:**
If you are new to the project, please familiarize yourself with the `README.md` file and the project structure before starting.

### PR DESCRIPTION
**Pull request by aninagotgit**
test: add unit tests
test: move contextLoads test to IT package
refactor: convert field injection to constructor injection in JwtAuthenticationHelper

Closes #26 

## Summary by CodeRabbit

* **Refactor**
  * Improved test class naming for clarity and consistency.

* **Style**
  * Updated dependency injection style for improved maintainability.

* **Tests**
  * Introduced comprehensive unit and integration tests covering analytics, books, borrowings, email, members, newsletters, notifications, authentication, and user management.
  * Added utility methods for generating test data and supporting reusable test setups.

### CODE REVIEW

File: src/it/java/com/libraryman_api/LibrarymanApiApplicationIntegrationTest.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import org.springframework.boot.test.context.SpringBootTest;
 
 @SpringBootTest
-class LibrarymanApiApplicationTests {
+class LibrarymanApiApplicationIntegrationTest {
```
Comment (aninagotgit) [inline src/it/java/com/libraryman_api/LibrarymanApiApplicationIntegrationTest.java:7]: Moved the test that requires the Spring application context to start up into a folder for Integration tests. (IT)
More integration tests can be added here.

File: src/main/java/com/libraryman_api/security/jwt/JwtAuthenticationHelper.java [hunk 2]
```diff
@@ -16,8 +16,11 @@
 public class JwtAuthenticationHelper {
 
     private static final long JWT_TOKEN_VALIDITY = 60 * 60;
-    @Value("${jwt.secretKey}")
-    private String secret;
+    private final String secret;
+
+    public JwtAuthenticationHelper(@Value("${jwt.secretKey}") String secret) {
+        this.secret = secret;
+    }
```
Comment (aninagotgit) [inline src/main/java/com/libraryman_api/security/jwt/JwtAuthenticationHelper.java:23]: Use constructor injection instead of field injection in order to write tests.

**Review by ajaynegi45 — APPROVED** [review]
Hey @aninagotgit 👋

Wow — what an incredible contribution! 🚀 A 2000+ line commit that not only brings in comprehensive unit and integration tests, but also modernizes our codebase. You've clearly put a lot of thought and care into improving both the test structure and overall maintainability. 🙌

It would be awesome if you could take a look at other present issues and push a fix when you get a chance. Let me know if you need help — happy to assist!

Also, if you're enjoying working on this repo, I’d love for you to explore some of the other Java projects I maintain — especially [Vayu](https://github.com/ajaynegi45/Vayu) 🌬️. It’s an evolving project with tons of interesting problems — could definitely use your eye for clean code and thoughtful testing!

Thanks again for this beast of a contribution. Can't wait to get this merged once those last bits are resolved. Keep rocking! 💪
